osu.Game/Overlays/RankingsOverlay.cs

```diff
@@ -144,100 +139,101 @@ public void ShowSpotlights()
             Show();
         }
 
-        private void loadNewContent()
+        private void selectDisplayToLoad()
         {
-            loading.Show();
-
+            performanceDisplay = null;
             cancellationToken?.Cancel();
-            lastRequest?.Cancel();
+            spotlightsRequest?.Cancel();
 
-            if (Scope.Value == RankingsScope.Spotlights)
+            loading.Show();
+
+            if (!API.IsLoggedIn)
             {
-                loadContent(new SpotlightsLayout
-                {
-                    Ruleset = { BindTarget = ruleset }
-                });
+                loadDisplayAsync(Empty());
                 return;
             }
 
-            var request = createScopedRequest();
-            lastRequest = request;
-
-            if (request == null)
+            if (Scope.Value == RankingsScope.Spotlights)
             {
-                loadContent(null);
+                loadSpotlightsDisplay();
                 return;
             }
 
-            request.Success += () => Schedule(() => loadContent(createTableFromResponse(request)));
-            request.Failure += _ => Schedule(() => loadContent(null));
-
-            api.Queue(request);
+            loadDisplayAsync(selectDisplay());
         }
 
-        private APIRequest createScopedRequest()
+        private void loadSpotlightsDisplay()
         {
-            switch (Scope.Value)
+            spotlightsRequest = new GetSpotlightsRequest();
+            spotlightsRequest.Success += response => Schedule(() =>
             {
-                case RankingsScope.Performance:
-                    return new GetUserRankingsRequest(ruleset.Value, country: Country.Value?.FlagName);
-
-                case RankingsScope.Country:
-                    return new GetCountryRankingsRequest(ruleset.Value);
+                var display = new SpotlightsRankingsDisplay
+                {
+                    Current = ruleset,
+                    StartLoading = loading.Show,
+                    FinishLoading = loading.Hide,
+                    Spotlights = response.Spotlights
+                };
+                loadDisplayAsync(display);
+            });
+            API.Queue(spotlightsRequest);
```

Comment:
this logic should ideally be in the spotlights display, not out here like this. the overlay should only be managing navigation between displays.

Reply:
Moved to spotlights display